### PR TITLE
feat: make github gateway webhook creation optional

### PR DESCRIPTION
**Pull request by gordonbondon**
A possible approach to #689.

### CODE REVIEW

**Comment by alexec** [comment]
@VaibhavPage @whynowy @gordonbondon do we still need this?

**Comment by whynowy** [comment]
@gordonbondon - I feel like this is a valid use case, could you rebase to latest code? Or close this PR and submit another one with clean base.

**Comment by gordonbondon** [comment]
Rebased!

File: eventsources/sources/github/start.go [hunk 1]
```diff
@@ -253,6 +256,15 @@ func (router *Router) PostInactivate() error {
 
 	githubEventSource := router.githubEventSource
 
+	if githubEventSource.APIToken == nil {
+		logger := router.route.Logger.With(map[string]interface{}{
```
Comment (whynowy) [inline eventsources/sources/github/start.go:260]: We changed to use `zap` as log tool, check the following code for `With` function.

File: eventsources/sources/github/start.go [hunk 2]
```diff
@@ -253,6 +256,15 @@ func (router *Router) PostInactivate() error {
 
 	githubEventSource := router.githubEventSource
 
+	if githubEventSource.APIToken == nil {
```
Comment (whynowy) [inline eventsources/sources/github/start.go:259]: Adding `|| githubEventSource.Webhook.URL == ""` might be better.

File: eventsources/sources/github/start.go (hunk 1 above)
Comment (gordonbondon) [reply eventsources/sources/github/start.go]: done

File: eventsources/sources/github/start.go (hunk 2 above)
Comment (gordonbondon) [reply eventsources/sources/github/start.go]: done

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**